abi/src/main/java/org/web3j/abi/TypeEncoder.java

```diff
@@ -326,4 +384,39 @@ private static String encodeDynamicStructValues(final DynamicStruct value) {
         }
         return result.toString();
     }
+
+    /**
+     * Checks if the received array doesn't contain any element that can make the array unsupported
+     * for abi.encodePacked
+     *
+     * @param value Array to which the abi.encodePacked should be applied
+     * @param <T> Types of elements from the array
+     * @return if the encodePacked is supported for the given array
+     */
+    private static <T extends Type> boolean isSupportingEncodedPacked(Array<T> value) {
+        if (Utf8String.class.isAssignableFrom(value.getComponentType())
```

Comment:
It looks like you make encode/decode Utf8String below. (DefaultFunctionEncoderTest.java: 92)

Reply:
indeed but here I check only for arrays if supports encodedPacked, because for example now encodedPacked is not supported for arrays of Utf8Strings, or other nested arrays